<launch>

  <!-- set params for initialization -->
  <param name="szenario" value="empty_world"/>
  <arg name="model" default="burger"/>
  <arg name="first_tb3"  default="cat"/>
  <arg name="second_tb3" default="mouse"/>

  <!-- cat start position -->
  <arg name="first_tb3_x_pos" default="-1.0"/>
  <arg name="first_tb3_y_pos" default=" 2.0"/>
  <arg name="first_tb3_z_pos" default=" 0.0"/>
  <arg name="first_tb3_yaw"   default=" 0.0"/>
  <param name="cat_start_pos_x" value="$(arg first_tb3_x_pos)"/>
  <param name="cat_start_pos_y" value="$(arg first_tb3_y_pos)"/>

  <!-- mouse start position -->
  <arg name="second_tb3_x_pos" default=" 2.0"/>
  <arg name="second_tb3_y_pos" default="-1.0"/>
  <arg name="second_tb3_z_pos" default=" 0.0"/>
  <arg name="second_tb3_yaw"   default=" 1.57"/>
  <param name="mouse_start_pos_x" value="$(arg second_tb3_x_pos)"/>
  <param name="mouse_start_pos_y" value="$(arg second_tb3_y_pos)"/>
  
  <!-- cheese position -->  
  <arg name="cheese_x_pos" default="-2.0"/>
  <arg name="cheese_y_pos" default=" 1.0"/>
  <param name="cheese_pos_x" value="$(arg cheese_x_pos)"/>
  <param name="cheese_pos_y" value="$(arg cheese_y_pos)"/>
  

  <!-- empty world simulation -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find catch)/worlds/empty.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>  

  <!-- cat -->
  <group ns = "$(arg first_tb3)">
    <param name="robot_description" command="$(find xacro)/xacro $(find catch)/urdf/turtlebot3_$(arg model)_cat.urdf.xacro" />
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg first_tb3)" />
    </node>
    
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg first_tb3) -x $(arg first_tb3_x_pos) -y $(arg first_tb3_y_pos) -z $(arg first_tb3_z_pos) -Y $(arg first_tb3_yaw) -param robot_description" />   
    
    <node name="cat_perception" type="perception.py" pkg="catch" args="/mouse/odom /cat/perception_of_mouse 0.2 0.2" /> 
  </group>

  
  <!-- mouse -->
  <group ns = "$(arg second_tb3)">
    <param name="robot_description" command="$(find xacro)/xacro $(find catch)/urdf/turtlebot3_$(arg model)_mouse.urdf.xacro" />

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg second_tb3)" />
    </node>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg second_tb3) -x $(arg second_tb3_x_pos) -y $(arg second_tb3_y_pos) -z $(arg second_tb3_z_pos) -Y $(arg second_tb3_yaw) -param robot_description" />
    
    <node name="mouse_perception" type="perception.py" pkg="catch" args="/cat/odom /mouse/perception_of_cat 0.2 0.2" /> 
  </group>


  <group ns = "cheese">
    <param name="robot_description" command="$(find xacro)/xacro $(find catch)/urdf/cheese.urdf.xacro" />
		
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model cheese1 -x $(arg cheese_x_pos) -y $(arg cheese_y_pos) -param robot_description" />
  </group> 	

  	
  <node name="refree" type="refree.py" pkg="catch"  />


  <include file="$(find gruppe2)/launch/gruppe2.launch"/>
  
  <!-- <include file="$(find group_y)/launch/team_mouse.launch"/> -->

  

  
</launch>
